arm-automation/2015-10-31/swagger/connection.json

```diff
@@ -0,0 +1,470 @@
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ConnectionListResult"
+            }
+          },
+		  "default": {
+            "description": "Automation error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ConnectionCreateOrUpdateProperties": {
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "Gets or sets the description of the connection."
+        },
+        "connectionType": {
+          "$ref": "#/definitions/ConnectionTypeAssociationProperty",
+          "description": "Gets or sets the connectionType of the connection."
+        },
+        "fieldDefinitionValues": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets the field definition properties of the connection."
+        }
+      },
+      "required": [
+        "connectionType"
+      ],
+      "description": "The properties of the create connection properties"
+    },
+    "ConnectionCreateOrUpdateParameters": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Gets or sets the name of the connection."
+        },
+        "properties": {
+          "$ref": "#/definitions/ConnectionCreateOrUpdateProperties",
+          "x-ms-client-flatten": true,
+          "description": "Gets or sets the properties of the connection."
+        }
+      },
+      "required": [
+        "name",
+        "properties"
+      ],
+      "description": "The parameters supplied to the create or update connection operation."
+    },
+    "ConnectionProperties": {
+      "properties": {
+        "connectionType": {
+          "$ref": "#/definitions/ConnectionTypeAssociationProperty",
+          "description": "Gets or sets the connectionType of the connection."
+        },
+        "fieldDefinitionValues": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets the field definition values of the connection."
+        },
+        "creationTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the creation time."
+        },
+        "lastModifiedTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the last modified time."
+        },
+        "description": {
+          "type": "string",
+          "description": "Gets or sets the description."
+        }
+      },
+      "description": "Definition of the connection properties."
+    },
+    "Connection": {
```

Comment:
Is this a tracked/Proxy Resource? Should it inherit from the common `Resource` model having `id`, `name`, `type` as readOnly?